### PR TITLE
Resolve chameleon GAGS item issues

### LINKED ISSUE #58782: GAGS jumpsuits and shoes broke chameleon shoes/jumpsuit sprites
## Round ID:
Tested locally

## Testmerges:
N/A

## Reproduction:
image
I am physically incapable of escaping this hell

### PR DESCRIPTION
**Pull request by celotajstg**
## About The Pull Request

This PR fixes issues with the chameleon jumpsuit and sneakers. In addition to that, it properly defines setting for GAGS worn and inhand icons instead of having them be in Initialize() for specific item types. Names for the prisoner jumpsuits and wheelys have also been reverted to how they were before the GAGS PR.

fixes #58782 

## Why It's Good For The Game

Brings back expected behavior for affected items.

## Changelog
:cl: Celotajs
fix: Chameleon sneakers and jumpsuits now work correctly
/:cl:

### CODE REVIEW

**Comment by celotajstg** [comment]
fixes #58782

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -265,6 +272,15 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
 	return
 
+/// Checks if this atom uses the GAS system and if so updates the worn and inhand icons
+/obj/item/proc/update_item_greyscale()
+	if(greyscale_config_worn && greyscale_colors)
+		worn_icon = SSgreyscale.GetColoredIconByType(greyscale_config_worn, greyscale_colors)
+	if(greyscale_config_inhand_left && greyscale_colors)
+		lefthand_file = SSgreyscale.GetColoredIconByType(greyscale_config_inhand_left, greyscale_colors)
+	if(greyscale_config_inhand_right && greyscale_colors)
+		righthand_file = SSgreyscale.GetColoredIconByType(greyscale_config_inhand_right, greyscale_colors)
```
Comment (LemonInTheDark) [inline code/game/objects/items.dm:282]: you should early return on greyscale colors, this is a bit messy

File: code/modules/clothing/chameleon.dm [hunk 2]
```diff
@@ -220,16 +220,28 @@
...
-		I.lefthand_file = initial(picked_item.lefthand_file)
-		I.righthand_file = initial(picked_item.righthand_file)
+		if(initial(picked_item.greyscale_config_inhand_left) && initial(picked_item.greyscale_colors))
+			I.lefthand_file = SSgreyscale.GetColoredIconByType(initial(picked_item.greyscale_config_inhand_left), initial(picked_item.greyscale_colors))
+		else
+			I.lefthand_file = initial(picked_item.lefthand_file)
+		if(initial(picked_item.greyscale_config_inhand_right) && initial(picked_item.greyscale_colors))
+			I.righthand_file = SSgreyscale.GetColoredIconByType(initial(picked_item.greyscale_config_inhand_right), initial(picked_item.greyscale_colors))
+		else
+			I.righthand_file = initial(picked_item.righthand_file)
```
Comment (LemonInTheDark) [inline code/modules/clothing/chameleon.dm:230]: kind of the same thing here, but it's more up to you in this instance

File: code/modules/clothing/chameleon.dm [hunk 3]
```diff
@@ -220,16 +220,28 @@
...
 		I.worn_icon_state = initial(picked_item.worn_icon_state)
 		if(istype(I, /obj/item/clothing) && istype(initial(picked_item), /obj/item/clothing))
 			var/obj/item/clothing/CL = I
 			var/obj/item/clothing/PCL = picked_item
 			CL.flags_cover = initial(PCL.flags_cover)
-	target.icon = initial(picked_item.icon)
+	if(initial(picked_item.greyscale_config) && initial(picked_item.greyscale_colors))
+		target.icon = SSgreyscale.GetColoredIconByType(initial(picked_item.greyscale_config), initial(picked_item.greyscale_colors))
+	else
+		target.icon = initial(picked_item.icon)
```
Comment (LemonInTheDark) [inline code/modules/clothing/chameleon.dm:244]: these apply to the above comment too

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

**Comment by celotajstg** [comment]
Done

File: code/game/objects/items.dm [hunk 4]
```diff
@@ -265,6 +272,17 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
+/// Checks if this atom uses the GAS system and if so updates the worn and inhand icons
+/obj/item/proc/update_item_greyscale()
+	if(greyscale_colors)
+		if(greyscale_config_worn)
+			worn_icon = SSgreyscale.GetColoredIconByType(greyscale_config_worn, greyscale_colors)
+		if(greyscale_config_inhand_left)
+			lefthand_file = SSgreyscale.GetColoredIconByType(greyscale_config_inhand_left, greyscale_colors)
+		if(greyscale_config_inhand_right)
+			righthand_file = SSgreyscale.GetColoredIconByType(greyscale_config_inhand_right, greyscale_colors)
+	return
```
Comment (LemonInTheDark) [inline code/game/objects/items.dm:284]: You want something more in line with this, to lower the amount of unneeded indents https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-early-return

Comment (celotajstg) [reply code/game/objects/items.dm]: I think I got it, please check

Comment (LemonInTheDark) [reply code/game/objects/items.dm]: yeah that's just fine, good work

**Pull request merged**